<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AppointmentView</class>
 <widget class="QWidget" name="AppointmentView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>预约排班管理</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="margin">
    <number>15</number>
   </property>
   <!-- 左侧：日历+筛选区（移除科室筛选） -->
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>15</number>
     </property>
     <!-- 日历控件（可视化选日期） -->
     <item>
      <widget class="QCalendarWidget" name="calendarWidget"/>
     </item>
     <!-- 筛选条件（仅保留医生筛选） -->
     <item>
      <widget class="QGroupBox" name="groupBoxFilter">
       <property name="title">
        <string>筛选条件</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="margin">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="labelDoctorFilter">
          <property name="text">
           <string>选择医生：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboDoctor">
          <item>
           <property name="text">
            <string>全部医生</string>
           </property>
          </item>
         </widget>
        </item>
        <!-- 新增：筛选状态（匹配数据库STATUS字段） -->
        <item>
         <widget class="QLabel" name="labelStatusFilter">
          <property name="text">
           <string>预约状态：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboStatusFilter">
          <item>
           <property name="text">
            <string>全部状态</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>未确认</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>已确认</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>已取消</string>
           </property>
          </item>
         </widget>
        </item>
        <!-- 新增：筛选按钮（触发筛选逻辑） -->
        <item>
         <widget class="QPushButton" name="btFilter">
          <property name="text">
           <string>筛选</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <!-- 右侧：日期选择+操作按钮+表格+状态修改按钮 -->
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <!-- 日期选择+增删改按钮 -->
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>10</number>
       </property>
       <item>
        <widget class="QLabel" name="labelDate">
         <property name="text">
          <string>选择日期：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDateEdit" name="dateEditSelected">
         <property name="calendarPopup">
          <bool>true</bool>
         </property>
         <!-- 默认选中当天 -->
         <property name="date">
          <date>
           <year>2025</year>
           <month>12</month>
           <day>27</day>
          </date>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer1">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btAdd">
         <property name="text">
          <string>新增预约</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btEdit">
         <property name="text">
          <string>编辑</string>
         </property>
         <!-- 默认禁用，选中记录后启用 -->
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btDelete">
         <property name="text">
          <string>删除</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <!-- 预约列表表格 -->
     <item>
      <widget class="QTableView" name="tableView">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <!-- 表格自动填充剩余空间 -->
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <!-- 状态修改按钮（确认/取消预约） -->
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="btConfirm">
         <property name="text">
          <string>确认预约</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btCancel">
         <property name="text">
          <string>取消预约</string>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <!-- 绑定日历选中日期到日期输入框 -->
  <connection>
   <sender>calendarWidget</sender>
   <signal>clicked(QDate)</signal>
   <receiver>dateEditSelected</receiver>
   <slot>setDate(QDate)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
